Add instructions for custom logging in native modules (#4447)
facebook/react-native-website: docs/debugging-native-code.md

You may also access these through Debug > Open System Log… in the iOS Simulator or by running `adb logcat "*:S" ReactNative:V ReactNativeJS:V` in a terminal while an Android app is running on a device or emulator.

> ### Custom Logs for Native Modules
> 
> If you are writing native modules and want to add custom logs to your module for debugging purposes, you can use the following method:
> 
> #### Android (Java/Kotlin)
> 
> In your native module, use the `Log` class to add logs that can be viewed in Logcat:
> 
> ```java
> import android.util.Log;
> 
> private void log(String message) {
>     Log.d("YourModuleName", message);
> }
> ```
> 
> To view these logs in Logcat, use this command:
> 
> ```shell
> adb logcat "*:S" ReactNative:V ReactNativeJS:V YourModuleName:D
> ```
> 
> Replace `YourModuleName` with the tag used in your logs.
> 
> #### iOS (Objective-C/Swift)
> 
> In your native module, use `NSLog` for custom logs:
> 
> ```objective-c
> NSLog(@"YourModuleName: %@", message);
> ```
> 
> Or, in Swift:
> 
> ```swift
> print("YourModuleName: \(message)")
> ```
> 
> These logs will appear in the Xcode console when running the app.
> 

Reviewer: Agree that this information reasonably generic, but am happy to include. Can we change this to a `<details>` block + shorten slightly, per the below?

```suggestion
<details>
<summary>**💡 Custom Native Logs**</summary>

If you are writing a Native Module and want to add custom logs to your module for debugging purposes, you can use the following method:

#### Android (Java/Kotlin)

In your native module, use the `Log` class to add logs that can be viewed in Logcat:

```java
import android.util.Log;

private void log(String message) {
    Log.d("YourModuleName", message);
}
```

To view these logs in Logcat, use this command, replacing `YourModuleName` with your custom tag:

```shell
adb logcat "*:S" ReactNative:V ReactNativeJS:V YourModuleName:D
```

#### iOS (Objective-C/Swift)

In your native module, use `NSLog` for custom logs:

```objective-c
NSLog(@"YourModuleName: %@", message);
```

Or, in Swift:

```swift
print("YourModuleName: \(message)")
```

These logs will appear in the Xcode console when running the app.

</details>

```
